tests/service/StructureTests.fs

```diff
@@ -301,6 +301,25 @@ match None with     // 2
          (6, 4, 10, 10), (5, 6, 10, 10)
          (6, 4, 10, 10), (6, 19, 10, 10)
          (9, 8, 10, 10), (8, 10, 10, 10) ]
+
+[<Test>]
+let ``matchbang``() =
+    """
+match! None with     // 2
+| Some _ ->          // 3
+    ()               // 4
+| None ->            // 5
+    match None with  // 6
+    | Some _ -> ()   // 7
+    | None ->        // 8
+        let x = ()   // 9
+        ()           // 10
```

Comment:
The example code is not realistical.

Reply:
This is just like the other tests. Do you have a suggested improvement?

Reply:
```fsharp
maybe {
    match! Some None with     
    | Some _ ->          
        ()               
    | None ->            
        match None with  
        | Some _ -> ()   
        | None ->        
            let x = ()   
            ()
}                       
```